templates/ValidationScript.ss

```diff
@@ -35,20 +44,76 @@
 						'{$Name.JS}': {$ValidationJSON.RAW},
 					<% end_if %><% end_if %>
 				<% end_loop %>
-			},
+			}
+
+			/* 
+			 * Conditional options.
+			 * Using leading commas so we don't get a trailing comma on
+			 * the last option. Trailing commas can break IE.
+			 */
 			<% if $EnableLiveValidation %>
 				// Enable live validation
-				onfocusout : function(element) { this.element(element); }
+				,onfocusout: function (element) { this.element(element); }
+			<% end_if %>
+
+			<% if $DisplayErrorMessagesAtTop %>
+				,invalidHandler: function (event, validator) {
+					var errorList = $('#error-container ul');
+
+					// Update the error list with errors from the validator.
+					// We do this because top messages are not part of the regular
+					// error message life cycle, which jquery.validate handles for us.
+					errorList.empty();
+
+					$.each(validator.errorList, function () {
+						applyTopErrorMessage(this.element.id + '-error-top', this.message);
+					});
+				}
+				,onfocusout: false
 			<% end_if %>
 		});
+
 		<% if $HideFieldLabels %>
 			// Hide field labels (use HTML5 placeholder instead)
-			$("#Form_Form label.left").each(function() {
+			$("#" + formId + "label.left").each(function() {
 				$("#"+$(this).attr("for"))
 					.attr("placeholder", $(this).text());
 				$(this).remove();
 			});
 			Placeholders.init();
 		<% end_if %>
+
+		<% if $DisplayErrorMessagesAtTop %>
+			/**
+			 * @applyTopErrorMessage
+			 * @param {string} elementId - The HTML id of the <li> where the error message is displayed.
+			 * @param {string} message - The error message to display.
+			 * @desc Update an error message (displayed at the top of the form).
+			 */
+			function applyTopErrorMessage(elementId, message) {
+				var errorContainer = $('#error-container'),
```

Comment:
I think we should make the error-container selector configurable. It would be best to put it into a yml config option (or just a private static on the UserDefinedForm class).